src/ucp/wireup/wireup.c

```diff
@@ -1431,21 +1394,30 @@ ucp_wireup_replace_wireup_msg_lane(ucp_ep_h ep, ucp_ep_config_key_t *key,
     old_wireup_ep = ucp_wireup_ep(ucp_ep_get_lane(ep, old_lane));
     ucs_assert_always(old_wireup_ep != NULL);
 
+    /* Select CM/non-reused lane as new wireup lane */
+    new_wireup_lane = ucp_ep_find_non_reused_lane(ep, key, reuse_lane_map);
+
     /* Set wireup EP for new configuration's wireup lane */
     if (ucp_ep_has_cm_lane(ep)) {
         /* Use existing EP from CM lane */
         new_wireup_ep = ucp_ep_get_cm_wireup_ep(ep);
-        ucs_assert(new_wireup_ep != NULL);
-    } else {
+    } else if (new_wireup_lane != UCP_NULL_LANE) {
         /* Create new EP for non-CM flow */
         status = ucp_wireup_ep_create(ep, &uct_ep);
         if (status != UCS_OK) {
             return status;
         }
 
         new_wireup_ep = ucp_wireup_ep(uct_ep);
+    } else {
+        /* Use existing EP from selected wireup_ep */
+        new_wireup_lane = ucp_ep_find_wireup_ep_lane(ep);
```

Comment:
if we get here, it means all lanes are reused, so maybe just keep using old_wireup_ep as the wireup ep? since it is reused as well
BTW, why not always keep using old_wireup_ep if it is reused? and create new wireup ep only if it is not reused

Reply:
old_wireup_ep is indeed selected if it is reused (in line 1500).
But consider the case where:
   1) old_wireup_ep is removed.
   2) all other lanes are reused.
In this case we neeed to select a new lane to hold the old_wireup_ep, even thogh all lanes in new config are reused.

Reply:
but if the lane that old_wireup_ep was part of is reused, why would it be removed? wouldn't it be reused?

Reply:
old_wireup_ep lane is not reused, it is discarded.
All the other lanes are reused.